Lib/test/test_functools.py

```diff
@@ -1876,8 +1876,7 @@ def fib(n):
 
         if not support.Py_DEBUG:
             with support.infinite_recursion():
-                fib(1250)
-
+                self.assertRaises(RecursionError, fib, 2000)
```

Comment:
I believe this violates the intention of the test -- it wants to ensure that we can recurse at least a minimum given number of times. Maybe we need both?

Reply:
We already have tests for maxing out the recursion limit, both for C and Python recursion, so we probably don't need to add any. But there is no harm in having more tests.